components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/ClaimMetadataManagementServiceImpl.java

```diff
@@ -648,6 +652,95 @@ public List<Claim> getMappedExternalClaimsForLocalClaim(String localClaimURI, St
     }
 
+    /**
+     * Validate and sync the claim profile properties.
+     *
+     * @param claimProperties Map of claim properties to be updated.
+     * @throws ClaimMetadataClientException If an invalid profile name is found.
+     */
+    private void validateAndSyncAttributeProfileProperties(Map<String, String> claimProperties)
+            throws ClaimMetadataClientException {
+
+        Set<String> allowedClaimProfiles = getAllowedClaimProfiles();
+
+        // Validate profile names and throw an exception if an invalid profile name is found.
+        for (Map.Entry<String, String> entry : claimProperties.entrySet()) {
+            if (!entry.getKey().startsWith(ClaimConstants.PROFILES_CLAIM_PROPERTY_PREFIX)) {
+                continue;
+            }
+
+           String[] profilePropertyKeyArray = entry.getKey().split("\\.");
+           if (profilePropertyKeyArray.length < 2 || !allowedClaimProfiles.contains(profilePropertyKeyArray[1])) {
+               throw new ClaimMetadataClientException(ERROR_CODE_INVALID_ATTRIBUTE_PROFILE);
+           }
+        }
+
+        String[] allowedProfilePropertyKeys = {
+                ClaimConstants.SUPPORTED_BY_DEFAULT_PROPERTY,
+                ClaimConstants.REQUIRED_PROPERTY,
+                ClaimConstants.READ_ONLY_PROPERTY
+        };
+        for (String propertyKey: allowedProfilePropertyKeys) {
+            syncAttributeProfileProperties(claimProperties, allowedClaimProfiles, propertyKey);
+        }
+    }
+
+    /**
+     * Remove the profile properties if the profile property value is the same as the global property value.
+     * If all the profile properties have the same value, change the global property value to the profile value.
+     *
+     * @param claimProperties      Map of claim properties to be updated.
+     * @param allowedClaimProfiles Set of allowed claim profiles.
+     * @param propertyKey          Property key to be updated.
+     */
+    private void syncAttributeProfileProperties(Map<String, String> claimProperties, Set<String> allowedClaimProfiles,
+                                                String propertyKey) {
+
+        String globalValue = claimProperties.get(propertyKey);
+        boolean allProfileValuesMatch = true;
+        String consistentProfileValue = null;
+
+        for (String profileName : allowedClaimProfiles) {
+            String profilePropertyKey = ClaimConstants.PROFILES_CLAIM_PROPERTY_PREFIX + profileName +
+                    ClaimConstants.CLAIM_PROFILE_PROPERTY_DELIMITER + propertyKey;
+            String profilePropertyValue = claimProperties.get(profilePropertyKey);
+
+            // Remove the profile property if it is the same as the global property.
+            if (StringUtils.equals(globalValue, profilePropertyValue)) {
+                claimProperties.remove(profilePropertyKey);
+                allProfileValuesMatch = false;
```

Comment:
`allProfileValuesMatch` The variable name looks misleading. Is it should be `allProfileValuesMisMatch`?

boolean allProfileValuesMisMatch = false; // Initialize assuming all are same.

if (StringUtils.equals(globalValue, profilePropertyValue)) {
              // Remove redundant information as it is reflected in global value.
                claimProperties.remove(profilePropertyKey);
} else {
      // Found a profile mismacth.
      allProfileValuesMisMatch = true;
}